test/test_reactant.jl

```diff
@@ -54,14 +67,136 @@ GPUArrays.allowscalar(true)
     @test iteration(r_simulation) == iteration(simulation)
     @test time(r_simulation) == time(simulation)
 
-    # Data looks right:
-    u, v, w = model.velocities
-    ru, rv, rw = r_model.velocities
+    @info "  After running 3 time steps:"
+    @show maximum(abs, parent(u))
+    @show maximum(abs, parent(v))
+    @show maximum(abs, parent(w))
 
     @show maximum(abs.(parent(u) .- parent(ru)))
     @show maximum(abs.(parent(v) .- parent(rv)))
     @show maximum(abs.(parent(w) .- parent(rw)))
-    @test_broken parent(u) ≈ parent(ru)
-    @test_broken parent(v) ≈ parent(rv)
-    @test_broken parent(w) ≈ parent(rw)
+
+    @test parent(u) ≈ parent(ru)
+    @test parent(v) ≈ parent(rv)
+    @test parent(w) ≈ parent(rw)
+
+    return nothing
+end
+
+function add_one!(f)
+    arch = architecture(f)
+    launch!(arch, f.grid, :xyz, _add_one!, f)
+    return f
+end
+
+@kernel function _add_one!(f)
+    i, j, k = @index(Global, NTuple)
+    @inbounds f[i, j, k] += 1
+end
+
+@testset "Reactanigans unit tests" begin
+    @info "Performing Reactanigans unit tests..."
+    arch = ReactantState()
+    grid = RectilinearGrid(arch; size=(4, 4, 4), extent=(1, 1, 1))
+    c = CenterField(grid)
+    @test parent(c) isa Reactant.ConcreteRArray
+
+    @info "  Testing field set! with a number..."
+    set!(c, 1)
+    @test all(c .≈ 1)
+
+    @info "  Testing simple kernel launch!..."
+    add_one!(c)
+    @test all(c .≈ 2)
+
+    set!(c, (x, y, z) -> x + y * z)
+    x, y, z = nodes(c)
+
+    @allowscalar begin
+        @test c[1, 1, 1] == x[1] + y[1] * z[1]
+        @test c[1, 2, 1] == x[1] + y[2] * z[1]
+        @test c[1, 2, 3] == x[1] + y[2] * z[3]
+    end
+
+    @jit fill_halo_regions!(c)
+
+    @allowscalar begin
+        @test c[1, 1, 0] == c[1, 1, 1]
+    end
+
+    d = CenterField(grid)
+    parent(d) .= 2
+
+    cd = Field(c * d)
+    compute!(cd)
+
+    @allowscalar begin
+        @test cd[1, 1, 1] == 2 * (x[1] + y[1] * z[1])
+        @test cd[1, 2, 1] == 2 * (x[1] + y[2] * z[1])
+        @test cd[1, 2, 3] == 2 * (x[1] + y[2] * z[3])
+    end
+end
+
+@testset "Reactant Super Simple Simulation Tests" begin
+    nonhydrostatic_model_kw = (; advection=WENO())
+    hydrostatic_model_kw = (; momentum_advection=WENO())
+    Nx, Ny, Nz = (10, 10, 10) # number of cells
+    halo = (7, 7, 7)
+    longitude = (0, 4)
+    latitude = (0, 4)
+    z = (-1, 0)
+    lat_lon_kw = (; size=(Nx, Ny, Nz), halo, longitude, latitude, z)
+    rectilinear_kw = (; size=(Nx, Ny, Nz), halo, x=(0, 1), y=(0, 1), z=(0, 1))
+
+    # FFTs are not supported by Reactant so we don't run this test:
```

Comment:
how are they not supported atm (I think regular FFTs ought to work)?

cc @mofeing @avik-pal

Reply:
They should be. What is the error that you are getting?

Reply:
I'll open a new PR right after this and we can address it